bundles/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/GroupRenderer.java

```diff
@@ -36,16 +38,27 @@ public boolean canRender(Widget w) {
      */
     @Override
     public EList<Widget> renderWidget(Widget w, StringBuilder sb) throws RenderException {
-        String snippet = getSnippet("group");
 
-        snippet = StringUtils.replace(snippet, "%id%", itemUIRegistry.getWidgetId(w));
-        snippet = StringUtils.replace(snippet, "%icon%", escapeURLPath(itemUIRegistry.getIcon(w)));
-        snippet = StringUtils.replace(snippet, "%label%", getLabel(w));
+        if (((Group) w).isExpanded()) {
+            EList<Widget> children = itemUIRegistry.getChildren((LinkableWidget) w);
 
-        // Process the color tags
-        snippet = processColor(w, snippet);
+            TextRenderer textRenderer = new TextRenderer();
+            textRenderer.setItemUIRegistry(itemUIRegistry);
+            
+            for (Widget child : children) 
+                textRenderer.renderWidget(child, sb);            
```

Comment:
Hm, the FrameRenderer does something very similar to what you want to do here, right? So I assume, instead of trying to render the children yourself, you should simply return them from this method.